diff --git a/src/celluloid-controller.c b/src/celluloid-controller.c
index c469638..6c1f482 100644
--- a/src/celluloid-controller.c
+++ b/src/celluloid-controller.c
@@ -586,7 +586,9 @@ is_active_handler(GObject *gobject, GParamSpec *pspec, gpointer data)
 static gboolean
 close_request_handler(CelluloidView *view, gpointer data)
 {
-	g_signal_emit_by_name(data, "shutdown");
+	CelluloidController *controller = CELLULOID_CONTROLLER(data);
+
+	celluloid_model_quit(controller->model);
 
 	return TRUE;
 }
diff --git a/src/celluloid-mpv.c b/src/celluloid-mpv.c
index 64d7671..c626919 100644
--- a/src/celluloid-mpv.c
+++ b/src/celluloid-mpv.c
@@ -761,10 +761,11 @@ celluloid_mpv_quit(CelluloidMpv *mpv)
 		priv->render_ctx = NULL;
 	}
 
-	g_assert(priv->mpv_ctx);
-	mpv_terminate_destroy(priv->mpv_ctx);
-
-	priv->mpv_ctx = NULL;
+	if (priv->mpv_ctx)
+	{
+		mpv_terminate_destroy(priv->mpv_ctx);
+		priv->mpv_ctx = NULL;
+	}
 }
 
 void
