--- a/transparency.js	2026-06-08 16:58:05.398149283 +0800
+++ b/transparency.js	2026-06-08 16:58:05.415876711 +0800
@@ -257,11 +257,7 @@
     this._dtpPanel.panel.set_style(
       'background: none; ' +
         'border-image: none; ' +
-        'background-image: none; ' +
-        this._gradientStyle +
-        this._borderStyle +
-        'transition-duration:' +
-        this.animationDuration,
+        'background-image: none; ',
     )
   }
 
--- a/ui/SettingsStyle.ui	2026-06-08 16:58:05.403402809 +0800
+++ b/ui/SettingsStyle.ui	2026-06-08 16:58:05.451919713 +0800
@@ -371,6 +371,7 @@
         </child>
         <child>
           <object class="AdwActionRow">
+            <property name="visible">False</property>
             <property name="title" translatable="yes">Override panel theme background opacity</property>
             <child>
               <object class="GtkSwitch" id="trans_opacity_override_switch">
@@ -381,6 +382,7 @@
         </child>
         <child>
           <object class="AdwActionRow" id="trans_opacity_box">
+            <property name="visible">False</property>
             <property name="title" translatable="yes">Panel background opacity (%)</property>
             <child>
               <object class="GtkSpinButton" id="trans_opacity_spinbutton">
@@ -394,6 +396,7 @@
         <child>
           <object class="AdwActionRow" id="trans_opacity_box2">
             <property name="subtitle" translatable="yes">Change opacity when a window gets close to the panel</property>
+            <property name="visible">False</property>
             <property name="title" translatable="yes">Dynamic background opacity</property>
             <child>
               <object class="GtkButton" id="trans_dyn_options_button">
@@ -420,6 +423,7 @@
     </child>
     <child>
       <object class="AdwPreferencesGroup" id="style_group_dynamic_trans3">
+        <property name="visible">False</property>
         <child>
           <object class="AdwActionRow">
             <property name="title" translatable="yes">Override panel theme gradient</property>
@@ -432,6 +436,7 @@
         </child>
         <child>
           <object class="AdwActionRow" id="trans_gradient_box">
+            <property name="visible">False</property>
             <property name="title" translatable="yes">Gradient top color and opacity (%)</property>
             <child>
               <object class="GtkColorButton" id="trans_gradient_color1_colorbutton">
@@ -450,6 +455,7 @@
         </child>
         <child>
           <object class="AdwActionRow" id="trans_gradient_box2">
+            <property name="visible">False</property>
             <property name="title" translatable="yes">Gradient bottom color and opacity (%)</property>
             <child>
               <object class="GtkColorButton" id="trans_gradient_color2_colorbutton">
