--- a/emojiSearchItem.js	2026-06-11 10:51:34.785318114 +0800
+++ b/emojiSearchItem.js	2026-06-11 10:51:43.288539990 +0800
@@ -44,6 +44,7 @@
 
     // initializing the "recently used" buttons
     this.recentlyUsedItem = this._recentlyUsedInit();
+    this.recentlyUsedItem.visible = false;
 
     this.searchEntry.clutter_text.connect("key-press-event", (actor, event) => {
       const symbol = event.get_key_symbol();
@@ -111,6 +112,7 @@
   _onSearchTextChanged() {
     let searchedText = this.searchEntry.get_text();
     if (searchedText === "") {
+      this.recentlyUsedItem.visible = false;
       this._buildRecents();
       this._updateSensitivity();
       return;
@@ -152,6 +154,7 @@
       }
     }
     this._updateSensitivity();
+    this.recentlyUsedItem.visible = firstEmptyIndex > 0;
   }
 
   _updateSensitivity() {
@@ -183,25 +186,12 @@
     return recentlyUsed;
   }
 
-  saveRecents() {
-    let backUp = [];
-    for (let i = 0; i < this._nbColumns; i++) {
-      backUp.push(this._recents[i].super_btn.label);
-    }
-    this._settings.set_strv("recently-used", backUp);
-  }
+  saveRecents() {}
+
 
   _buildRecents() {
-    let temp = this._settings.get_strv("recently-used");
     for (let i = 0; i < this._nbColumns; i++) {
-      if (i < temp.length) {
-        this._recents[i].super_btn.label = temp[i];
-      } else {
-        // If the extension was previously set with less "recently used
-        // emojis", we still need to load something in the labels.
-        // It will be a penguin for obvious reasons.
-        this._recents[i].super_btn.label = "🐧";
-      }
+      this._recents[i].super_btn.label = "";
     }
   }
 
@@ -214,21 +204,6 @@
     });
   }
 
-  shiftFor(currentEmoji) {
-    if (currentEmoji == "") return;
-    let temp = this._settings.get_strv("recently-used");
-    for (let i = 0; i < temp.length; i++) {
-      if (temp[i] == currentEmoji) {
-        temp.splice(i, 1);
-      }
-    }
-    for (let j = temp.length; j > 0; j--) {
-      temp[j] = temp[j - 1];
-    }
-    temp[0] = currentEmoji;
-    this._settings.set_strv("recently-used", temp);
-    this._buildRecents();
-    this.saveRecents();
-    this.emojiCopy._onSearchTextChanged();
-  }
+  shiftFor(currentEmoji) {}
+
 }
