--- a/emojiOptionsBar.js	2026-06-11 20:14:05.190849317 +0800
+++ b/emojiOptionsBar.js	2026-06-11 20:14:31.601138947 +0800
@@ -76,7 +76,7 @@
   // Update buttons appearance, reflecting the current state of the settings
   update() {
     this.removeCircle();
-    // this._toneArray[this._settings.get_int("skin-tone")].set_checked(true);
+    this._toneArray[this._settings.get_int("skin-tone")].set_checked(true);
     this._genderArray.forEach(function (b) {
       if (b) { // index 0 contains null
         b.set_checked(false);
