--- a/subprojects/extensions-app/data/ui/extension-row.ui
+++ b/subprojects/extensions-app/data/ui/extension-row.ui
@@ -177,12 +177,6 @@
                               <object class="GtkBox">
                                 <property name="orientation">vertical</property>
                                 <child>
-                                  <object class="GtkImage">
-                                    <property name="icon-name">website-symbolic</property>
-                                    <property name="accessible-role">presentation</property>
-                                  </object>
-                                </child>
-                                <child>
                                   <object class="GtkLabel">
                                     <property name="use-underline">true</property>
                                     <property name="label" translatable="yes">_Website</property>
@@ -204,12 +198,6 @@
                               <object class="GtkBox">
                                 <property name="orientation">vertical</property>
                                 <child>
-                                  <object class="GtkImage">
-                                    <property name="icon-name">settings-symbolic</property>
-                                    <property name="accessible-role">presentation</property>
-                                  </object>
-                                </child>
-                                <child>
                                   <object class="GtkLabel">
                                     <property name="use-underline">true</property>
                                     <property name="label" translatable="yes">_Settings</property>
