diff -ruN -x wallpapers -x po -x kcms -x runners -x kioworkers pw-orig/applets/notifications/components/JobItem.qml pw-pat/applets/notifications/components/JobItem.qml
--- pw-orig/applets/notifications/components/JobItem.qml	2026-07-14 19:00:12.000000000 +0800
+++ pw-pat/applets/notifications/components/JobItem.qml	2026-07-31 21:47:56.623261480 +0800
@@ -31,7 +31,7 @@
 
     Notifications.FileInfo {
         id: fileInfo
-        url: jobItem.totalFiles === 1 ? jobItem.modelInterface.jobDetails.effectiveDestUrl : ""
+        url: jobItem.totalFiles === 1 ? (jobItem.modelInterface.jobDetails?.effectiveDestUrl ?? "") : ""
     }
 
     SpeedChart {
@@ -114,7 +114,7 @@
         // We want the actions to be right-aligned but Row also reverses
         // the order of items, so we put them in reverse order
         layoutDirection: Qt.RightToLeft
-        visible: jobItem.modelInterface.jobDetails.effectiveDestUrl.toString() !== "" && !fileInfo.error
+        visible: (jobItem.modelInterface.jobDetails?.effectiveDestUrl?.toString() ?? "") !== "" && !fileInfo.error
 
         PlasmaComponents3.Button {
             id: otherFileActionsButton
@@ -141,7 +141,7 @@
 
             Notifications.FileMenu {
                 id: otherFileActionsMenu
-                url: jobItem.modelInterface.jobDetails.effectiveDestUrl
+                url: jobItem.modelInterface.jobDetails?.effectiveDestUrl ?? ""
                 onActionTriggered: action => jobItem.modelInterface.fileActionInvoked(action)
             }
         }
@@ -151,7 +151,10 @@
             width: Math.min(implicitWidth, jobItem.width - otherFileActionsButton.width - fileActionsRow.spacing)
             height: Math.max(implicitHeight, otherFileActionsButton.implicitHeight)
             text: i18nd("plasma_applet_org.kde.plasma.notifications", "Open")
-            onClicked: jobItem.modelInterface.openUrl(jobItem.modelInterface.jobDetails.effectiveDestUrl)
+            onClicked: {
+                const u = jobItem.modelInterface.jobDetails?.effectiveDestUrl ?? "";
+                fileInfo.openFolderUrl(u);
+            }
 
             states: [
                 State {
diff -ruN -x wallpapers -x po -x kcms -x runners -x kioworkers pw-orig/applets/notifications/fileinfo.cpp pw-pat/applets/notifications/fileinfo.cpp
--- pw-orig/applets/notifications/fileinfo.cpp	2026-07-14 19:00:12.000000000 +0800
+++ pw-pat/applets/notifications/fileinfo.cpp	2026-07-31 21:47:56.608794059 +0800
@@ -187,3 +187,13 @@
 }
 
 #include "moc_fileinfo.cpp"
+
+#include <QProcess>
+
+void FileInfo::openFolderUrl(const QUrl &url)
+{
+    if (url.isValid() && !url.isEmpty() && url.isLocalFile()) {
+        QProcess::startDetached(QStringLiteral("dolphin"),
+            {QStringLiteral("--new-window"), url.toLocalFile()});
+    }
+}
diff -ruN -x wallpapers -x po -x kcms -x runners -x kioworkers pw-orig/applets/notifications/fileinfo.h pw-pat/applets/notifications/fileinfo.h
--- pw-orig/applets/notifications/fileinfo.h	2026-07-14 19:00:12.000000000 +0800
+++ pw-pat/applets/notifications/fileinfo.h	2026-07-31 21:47:56.607496220 +0800
@@ -64,6 +64,8 @@
     QString openActionIconName() const;
     Q_SIGNAL void openActionIconNameChanged();
 
+    Q_INVOKABLE void openFolderUrl(const QUrl &url);
+
 private:
     void reload();
     void mimeTypeFound(const QString &mimeType);
