diff --git a/cosmic-settings/src/pages/power/backend/mod.rs b/cosmic-settings/src/pages/power/backend/mod.rs
--- a/cosmic-settings/src/pages/power/backend/mod.rs
+++ b/cosmic-settings/src/pages/power/backend/mod.rs
@@ -58,14 +58,13 @@
 pub enum PowerProfile {
     Battery,
     Balanced,
-    Performance,
 }
 
 impl PowerProfile {
     fn from_string(s: &str) -> PowerProfile {
         match s {
             "Battery" | "power-saver" => Self::Battery,
-            "Performance" | "performance" => Self::Performance,
+            "Performance" | "performance" => Self::Balanced,
             _ => Self::Balanced,
         }
     }
@@ -74,7 +73,6 @@
         match self {
             Self::Battery => fl!("power-mode", "battery"),
             Self::Balanced => fl!("power-mode", "balanced"),
-            Self::Performance => fl!("power-mode", "performance"),
         }
     }
 
@@ -82,7 +80,6 @@
         match self {
             Self::Battery => fl!("power-mode", "battery-desc"),
             Self::Balanced => fl!("power-mode", "balanced-desc"),
-            Self::Performance => fl!("power-mode", "performance-desc"),
         }
     }
 }
@@ -91,7 +88,6 @@
     vec![
         PowerProfile::Battery,
         PowerProfile::Balanced,
-        PowerProfile::Performance,
     ]
 }
 
@@ -114,10 +110,6 @@
                 Ok(()) => tracing::info!("Balanced mode activated."),
                 Err(e) => tracing::error!("{e}"),
             },
-            PowerProfile::Performance => match daemon.performance().await {
-                Ok(()) => tracing::info!("Performance mode activated."),
-                Err(e) => tracing::error!("{e}"),
-            },
         }
     }
 }
@@ -181,10 +173,6 @@
                 Ok(()) => tracing::info!("Balanced mode activated."),
                 Err(e) => tracing::error!("{e}"),
             },
-            PowerProfile::Performance => match daemon.set_active_profile("performance").await {
-                Ok(()) => tracing::info!("Performance mode activated."),
-                Err(e) => tracing::error!("{e}"),
-            },
         }
     }
 }
