diff --git a/src/gui/gcal-window.blp b/src/gui/gcal-window.blp
index 9c058e2..5458fce 100644
--- a/src/gui/gcal-window.blp
+++ b/src/gui/gcal-window.blp
@@ -384,14 +384,6 @@
       label: _("_Online Accounts…");
       action: "win.open-online-accounts";
     }
-
-    submenu {
-      label: _("Weather");
-
-      item {
-        custom: "weather";
-      }
-    }
   }
 
   section {
@@ -410,6 +402,4 @@
 PopoverMenu win_menu {
   menu-model: menu;
 
-  [weather]
-  $GcalWeatherSettings weather_settings {}
 }
diff --git a/src/gui/gcal-window.c b/src/gui/gcal-window.c
index 62f1092..f8af1f2 100644
--- a/src/gui/gcal-window.c
+++ b/src/gui/gcal-window.c
@@ -38,7 +38,6 @@
 #include "gcal-timeline.h"
 #include "gcal-timeline-subscriber.h"
 #include "gcal-view.h"
-#include "gcal-weather-settings.h"
 #include "gcal-week-view.h"
 #include "gcal-window.h"
 
@@ -146,7 +145,6 @@
   gint                open_edit_dialog_timeout_id;
 
   /* weather management */
-  GcalWeatherSettings *weather_settings;
 
   /* temp to keep event_creation */
   gboolean            open_edit_dialog;
@@ -1220,7 +1218,6 @@
    * these properties using the GtkBuilder .ui file.
    */
   g_object_bind_property (self, "context", self->calendars_list, "context", G_BINDING_DEFAULT | G_BINDING_SYNC_CREATE);
-  g_object_bind_property (self, "context", self->weather_settings, "context", G_BINDING_DEFAULT | G_BINDING_SYNC_CREATE);
   g_object_bind_property (self, "context", self->week_view, "context", G_BINDING_DEFAULT | G_BINDING_SYNC_CREATE);
   g_object_bind_property (self, "context", self->month_view, "context", G_BINDING_DEFAULT | G_BINDING_SYNC_CREATE);
   g_object_bind_property (self, "context", self->agenda_view, "context", G_BINDING_DEFAULT | G_BINDING_SYNC_CREATE);
@@ -1362,7 +1359,6 @@
   g_type_ensure (GCAL_TYPE_QUICK_ADD_POPOVER);
   g_type_ensure (GCAL_TYPE_SYNC_INDICATOR);
   g_type_ensure (GCAL_TYPE_SEARCH_BUTTON);
-  g_type_ensure (GCAL_TYPE_WEATHER_SETTINGS);
   g_type_ensure (GCAL_TYPE_WEEK_VIEW);
   g_type_ensure (GCAL_TYPE_DROP_OVERLAY);
 
@@ -1420,7 +1416,6 @@
   gtk_widget_class_bind_template_child (widget_class, GcalWindow, sync_indicator);
   gtk_widget_class_bind_template_child (widget_class, GcalWindow, search_button);
   gtk_widget_class_bind_template_child (widget_class, GcalWindow, views_stack);
-  gtk_widget_class_bind_template_child (widget_class, GcalWindow, weather_settings);
   gtk_widget_class_bind_template_child (widget_class, GcalWindow, week_view);
   gtk_widget_class_bind_template_child (widget_class, GcalWindow, drop_overlay);
   gtk_widget_class_bind_template_child (widget_class, GcalWindow, drop_target);
diff --git a/src/gui/meson.build b/src/gui/meson.build
index c706c0c..dfe3b81 100644
--- a/src/gui/meson.build
+++ b/src/gui/meson.build
@@ -18,7 +18,6 @@
     'gcal-quick-add-popover.blp',
     'gcal-search-button.blp',
     'gcal-sync-indicator.blp',
-    'gcal-weather-settings.blp',
     'gcal-window.blp',
     'shortcuts-dialog.blp',
   ),
@@ -55,6 +54,5 @@
   'gcal-search-button.c',
   'gcal-sync-indicator.c',
   'gcal-tests.c',
-  'gcal-weather-settings.c',
   'gcal-window.c',
 )
diff --git a/src/gui/gui.gresource.xml b/src/gui/gui.gresource.xml
index 46bb0b8..1f11a00 100644
--- a/src/gui/gui.gresource.xml
+++ b/src/gui/gui.gresource.xml
@@ -12,7 +12,6 @@
     <file compressed="true">gcal-quick-add-popover.ui</file>
     <file compressed="true">gcal-search-button.ui</file>
     <file compressed="true">gcal-sync-indicator.ui</file>
-    <file compressed="true">gcal-weather-settings.ui</file>
     <file compressed="true">gcal-window.ui</file>
   </gresource>
 </gresources>
