diff --git a/include/modules/sni/item.hpp b/include/modules/sni/item.hpp
index 503ab637..b943e4c5 100644
--- a/include/modules/sni/item.hpp
+++ b/include/modules/sni/item.hpp
@@ -12,6 +12,7 @@
 #include <sigc++/trackable.h>
 
 #include <set>
+#include <string>
 #include <string_view>
 
 #include "bar.hpp"
@@ -86,6 +87,8 @@ class Item : public sigc::trackable {
   gdouble distance_scrolled_y_ = 0;
   // visibility of items with Status == Passive
   bool show_passive_ = false;
+  // list of ignored item IDs
+  std::set<std::string> ignored_items_;
 
   const Bar& bar_;
 
diff --git a/man/waybar-tray.5.scd b/man/waybar-tray.5.scd
index dec5347f..1473532e 100644
--- a/man/waybar-tray.5.scd
+++ b/man/waybar-tray.5.scd
@@ -21,6 +21,10 @@ Addressed by *tray*
 	default: false ++
 	Defines visibility of the tray icons with *Passive* status.
 
+*ignore*: ++
+	typeof: array ++
+	List of application IDs to be hidden from the tray.
+
 *smooth-scrolling-threshold*: ++
 	typeof: double ++
 	Threshold to be used when scrolling.
@@ -48,6 +52,7 @@ Addressed by *tray*
 "tray": {
 	"icon-size": 21,
 	"spacing": 10,
+	// "ignore": ["pavucontrol", "blueman"],
   "icons": {
     "blueman": "bluetooth",
     "TelegramDesktop": "$HOME/.local/share/icons/hicolor/16x16/apps/telegram.png"
diff --git a/src/modules/sni/item.cpp b/src/modules/sni/item.cpp
index ef2543b5..69a43fa3 100644
--- a/src/modules/sni/item.cpp
+++ b/src/modules/sni/item.cpp
@@ -53,6 +53,11 @@ Item::Item(const std::string& bn, const std::string& op, const Json::Value& conf
   if (config["show-passive-items"].isBool()) {
     show_passive_ = config["show-passive-items"].asBool();
   }
+  if (config["ignore"].isArray()) {
+    for (const auto& item : config["ignore"]) {
+      ignored_items_.insert(item.asString());
+    }
+  }
 
   auto& window = const_cast<Bar&>(bar).window;
   window.signal_configure_event().connect_notify(sigc::mem_fun(*this, &Item::onConfigure));
@@ -152,6 +157,10 @@ void Item::setProperty(const Glib::ustring& name, Glib::VariantBase& value) {
     } else if (name == "Id") {
       id = get_variant<std::string>(value);
 
+      if (ignored_items_.find(id) != ignored_items_.end()) {
+        event_box.set_visible(false);
+      }
+
       /*
        * HACK: Electron apps seem to have the same ID, but tooltip seems correct, so use that as ID
        * to pass as the custom icon option. I'm avoiding being disruptive and setting that to the ID
