diff --git a/app/linux/CMakeLists.txt b/app/linux/CMakeLists.txt
--- a/app/linux/CMakeLists.txt
+++ b/app/linux/CMakeLists.txt
@@ -8,7 +8,7 @@
 set(BINARY_NAME "localsend_app")
 # The unique GTK application identifier for this application. See:
 # https://wiki.gnome.org/HowDoI/ChooseApplicationID
-set(APPLICATION_ID "org.localsend.localsend_app")
+set(APPLICATION_ID "localsend")
 
 # Explicitly opt in to modern CMake behaviors to avoid warnings with recent
 # versions of CMake.
diff --git a/app/linux/main.cc b/app/linux/main.cc
--- a/app/linux/main.cc
+++ b/app/linux/main.cc
@@ -1,6 +1,7 @@
 #include "my_application.h"
 
 int main(int argc, char** argv) {
+  g_set_prgname("localsend");
   g_autoptr(MyApplication) app = my_application_new();
   return g_application_run(G_APPLICATION(app), argc, argv);
 }
diff --git a/app/linux/my_application.cc b/app/linux/my_application.cc
--- a/app/linux/my_application.cc
+++ b/app/linux/my_application.cc
@@ -55,6 +55,7 @@
   }
 
   gtk_window_set_default_size(window, 400, 500);
+  gtk_window_set_icon_name(GTK_WINDOW(window), "localsend");
   gtk_widget_show(GTK_WIDGET(window));
 
   g_autoptr(FlDartProject) project = fl_dart_project_new();
