--- a/src/app.rs
+++ b/src/app.rs
@@ -943,6 +943,21 @@
                                 .height(Length::Fixed(32.0))
                                 .into(),
                         );
+                    } else if !self.alt_tab && item.name == "Toggle Mute" {
+                        let speaker_icon = icon::from_name("audio-speakers")
+                            .prefer_svg(true)
+                            .size(64)
+                            .fallback(Some(IconFallback::Names(vec![
+                                "application-default".into(),
+                                "application-x-executable".into(),
+                            ])))
+                            .handle();
+                        button_content.push(
+                            icon(speaker_icon)
+                                .width(Length::Fixed(32.0))
+                                .height(Length::Fixed(32.0))
+                                .into(),
+                        );
                     }
 
                     button_content.push(column![name, desc].width(Length::FillPortion(5)).into());
