--- a/src/app.rs
+++ b/src/app.rs
@@ -576,10 +576,17 @@
                                      // Fetch icon by name
-                                    IconSource::Name(name) => icon::from_name(&**name)
-                                        .prefer_svg(true)
-                                        .size(64)
-                                        .fallback(Some(IconFallback::Names(vec![
-                                            "application-default".into(),
-                                            "application-x-executable".into(),
-                                        ])))
-                                        .handle(),
+                                    IconSource::Name(name) => {
+                                        let name = if name.contains("muted") {
+                                            std::borrow::Cow::Owned("audio-speakers".into())
+                                        } else {
+                                            std::borrow::Cow::Borrowed(&**name)
+                                        };
+                                        icon::from_name(&*name)
+                                            .prefer_svg(true)
+                                            .size(64)
+                                            .fallback(Some(IconFallback::Names(vec![
+                                                "application-default".into(),
+                                                "application-x-executable".into(),
+                                            ])))
+                                            .handle()
+                                    }
                                      // By mime
