diff --git a/src/app.rs b/src/app.rs
index 5e9fb56..c60acb2 100644
--- a/src/app.rs
+++ b/src/app.rs
@@ -299,6 +299,10 @@
     !modifiers.alt() && !modifiers.logo()
 }
 
+fn strip_source(s: &str) -> &str {
+    s.split(" - ").nth(1).unwrap_or("")
+}
+
 async fn launch(
     token: Option<String>,
     app_id: String,
@@ -865,11 +869,12 @@
                 .iter()
                 .enumerate()
                 .flat_map(|(i, item)| {
-                    let (name, desc) = if item.window.is_some() {
-                        (&item.description, &item.name)
+                    let (name, desc_buf) = if item.window.is_some() {
+                        (item.description.clone(), item.name.clone())
                     } else {
-                        (&item.name, &item.description)
+                        (item.name.clone(), item.description.clone())
                     };
+                    let desc = strip_source(&desc_buf);
 
                     let name = Column::with_children(name.lines().map(|line| {
                         text::body(line.to_string())
