diff --git a/src/shell/element/window.rs b/src/shell/element/window.rs
index dd45de4b..8bd398d4 100644
--- a/src/shell/element/window.rs
+++ b/src/shell/element/window.rs
@@ -1012,7 +1012,6 @@ impl PointerTarget<State> for CosmicWindow {
 
                 let cursor_state = seat.user_data().get::<CursorState>().unwrap();
                 cursor_state.lock().unwrap().set_shape(next.cursor_shape());
-                seat.set_cursor_image_status(CursorImageStatus::default_named());
             }
         });
 
@@ -1036,7 +1035,6 @@ impl PointerTarget<State> for CosmicWindow {
 
                 let cursor_state = seat.user_data().get::<CursorState>().unwrap();
                 cursor_state.lock().unwrap().set_shape(next.cursor_shape());
-                seat.set_cursor_image_status(CursorImageStatus::default_named());
             }
         });
 
diff --git a/src/shell/element/stack.rs b/src/shell/element/stack.rs
index d2f9c949..d013233c 100644
--- a/src/shell/element/stack.rs
+++ b/src/shell/element/stack.rs
@@ -1578,7 +1578,6 @@ impl PointerTarget<State> for CosmicStack {
                 .lock()
                 .unwrap();
             cursor_state.set_shape(next.cursor_shape());
-            seat.set_cursor_image_status(CursorImageStatus::default_named());
         });
 
         event.location -= self.0.with_program(|p| {
@@ -1607,7 +1606,6 @@ impl PointerTarget<State> for CosmicStack {
                 .lock()
                 .unwrap();
             cursor_state.set_shape(next.cursor_shape());
-            seat.set_cursor_image_status(CursorImageStatus::default_named());
         });
 
         let active_window_geo = self.0.with_program(|p| {
