--- a/desktoppackage/contents/lockscreen/LockScreenUi.qml
+++ b/desktoppackage/contents/lockscreen/LockScreenUi.qml
@@ -55,9 +55,11 @@
 
         function onSucceeded() {
             if (authenticator.hadPrompt) {
+                mainStack.clear();
+                lockScreenRoot.uiVisible = false;
                 Qt.quit();
             } else {
                 mainStack.replace(null, Qt.resolvedUrl("NoPasswordUnlock.qml"),
                     {
                         userListModel: users
                     },
@@ -134,7 +136,6 @@
             } else if (uiVisible) {
                 fadeoutTimer.restart();
             }
-            authenticator.startAuthenticating();
         }
         onBlockUIChanged: {
             if (blockUI) {
@@ -195,7 +196,7 @@
             duration: Kirigami.Units.veryLongDuration * 2
         }
 
-        Component.onCompleted: launchAnimation.start();
+        Component.onCompleted: { launchAnimation.start(); authenticator.startAuthenticating(); }
 
         WallpaperFader {
             anchors.fill: parent
