diff --git a/src/resources/ui/nautilus-files-view-context-menus.ui b/src/resources/ui/nautilus-files-view-context-menus.ui
index bef413814..bbd1276ac 100644
--- a/src/resources/ui/nautilus-files-view-context-menus.ui
+++ b/src/resources/ui/nautilus-files-view-context-menus.ui
@@ -212,11 +212,6 @@
         <attribute name="action">view.compress</attribute>
         <attribute name="hidden-when">action-disabled</attribute>
       </item>
-      <item>
-        <attribute name="label" translatable="yes">Set as Background…</attribute>
-        <attribute name="action">view.set-as-wallpaper</attribute>
-        <attribute name="hidden-when">action-disabled</attribute>
-      </item>
       <item>
         <attribute name="label" translatable="yes" comments="Translators: This is the transitive verb meaning 'to send via email' (e.g. 'email this document to Angela).">Email…</attribute>
         <attribute name="action">view.send-email</attribute>
